src/utils/Rotator.ts

```diff
@@ -0,0 +1,544 @@
+  }
+
+  /**
+   * Update rect
+   */
+  updateRect(store: boolean) {
+    const el = this.el!;
+    const rotator = this;
+    const config = this.opts;
+    const rect = this.rectDim!;
+    const updateTarget = this.updateTarget;
+
+    // Use custom updating strategy if requested
+    if (isFunction(updateTarget)) {
+      updateTarget(el, rect, {
+        store,
+        rotator,
+        config,
+      });
+    } else {
+      const elStyle = el.style as Record<string, any>;
+      elStyle.rotate = rect.r + 'deg';
+    }
+
+    this.updateContainer();
+  }
+
+  updateContainer(opt: { forceShow?: boolean } = {}) {
+    const { opts, container, el } = this;
+    const { style } = container!;
+
+    if (!opts.avoidContainerUpdate && el) {
+      // On component rotate container fits the tool,
+      // to check if this update is required somewhere else point
+      // const toUpdate = ['rotate'];
+      // const rectEl = this.getElementPos(el, { target: 'container' });
+      // toUpdate.forEach(pos => (style[pos] = `${rectEl[pos]}px`));
+      if (opt.forceShow) style.display = 'block';
+    }
+
+    this.onUpdateContainer?.({
+      el: container!,
+      rotator: this,
+      opts: {
+        ...opts,
+        ...opt,
+      },
+    });
+  }
+
+  /**
+   * Handle ESC key
+   * @param  {Event} e
+   */
+  handleKeyDown(e: Event) {
+    // @ts-ignore
+    if (e.keyCode === 27) {
+      // Rollback to initial rotation
+      this.rectDim = this.startDim;
+      this.stop(e);
+    }
+  }
+
+  /**
+   * Handle mousedown to check if it's possible to start rotating
+   * @param  {Event} e
+   */
+  handleMouseDown(e: Event) {
+    const el = e.target as HTMLElement;
+    if (this.isHandler(el)) {
+      this.selectedHandler = el;
+      this.start(e);
+    } else if (el !== this.el) {
+      delete this.selectedHandler;
+      this.blur();
+    }
+  }
+
+  /**
+   * All positioning logic
+   * @return {Object}
+   */
+  calc(data: Rotator): RectDim | undefined {
+    const startDim = this.startDim!;
+    const deltaX = data.delta!.x;
+    const deltaY = data.delta!.y;
+    const box: RectDim = {
+      l: startDim.l,
+      t: startDim.t,
+      w: startDim.w,
+      h: startDim.h,
+      r: startDim.r,
+    };
+
+    if (!data) return;
+
+    const angle = (Math.atan2(deltaY, deltaX) * 180 / Math.PI) + 90;
+    
+    box.r = angle;
+
+    const snappingPoints = this.defOpts.snapPoints!;
+    const snappingOffset = this.defOpts.snapOffset!;
+
+    const prevSnappingPoint = Math.round(angle / snappingPoints) * snappingPoints;
+    const nextSnappingPoint = prevSnappingPoint + snappingPoints;
+    const closestSnappingPoint =
+      angle - prevSnappingPoint > nextSnappingPoint - angle ? nextSnappingPoint : prevSnappingPoint;
+
+    const isWithinSnapRange = Math.abs(angle - closestSnappingPoint) < snappingOffset;
+
+    // Enforce rotation snapping (unless shift key is being held)
+    const shouldSnap = !data.keys!.shift && isWithinSnapRange;
```

Comment:
Wil je dit niet voor de berekening van snapping doen? en dan als het false is direct box returnen?

Reply:
Fair enough